### PR TITLE
Add support for kind dual-stack

### PR DESCRIPTION
**Pull request by Kmoneal**
First steps in https://github.com/istio/istio/issues/37533.

* Configure cluster IP family to support dual-stack
* Configure coredns to support dual-stack

### CODE REVIEW

File: files/common/scripts/kind_provisioner.sh [hunk 1]
```diff
@@ -191,6 +191,44 @@ EOF
...
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  verbs:
+  - get
+EOF
+    kubectl patch deployment coredns -n kube-system -p '{"spec":{"template":{"spec":{"containers":[{"name":"coredns","readinessProbe":{"httpGet":{"path":"/health","port":8080}}}]}}}}'
+    kubectl patch deployment coredns -n kube-system -p '{"spec":{"template":{"spec":{"containers":[{"name":"coredns", "image":"coredns/coredns:1.8.7"}]}}}}'
+    kubectl rollout restart -n kube-system deployment coredns
```
Comment (howardjohn) [inline files/common/scripts/kind_provisioner.sh:229]: nit: this restarts 3 times. We can definitely remove the last one, but could also merge the first one?

**Review by howardjohn — COMMENTED** [review]
are the coredns changes upstream in some version?

**Comment by jacob-delgado** [comment]
> are the coredns changes upstream in some version?

CoreDNS 1.8.1+ is what fixed dual-stack support. See https://coredns.io/2021/01/20/coredns-1.8.1-release/

kind uses the default version of CoreDNS that is shipped with kubeadm. kubeadm did not use a version of CoreDNS that support dual-stack services until 1.22.0. We had to support previous versions of K8s pre-1.22 that did not contain an updated CoreDNS changes. @Kmoneal please remove the patches for CoreDNS as in open source we are only planning on supporting 1.22+ for dual-stack.

File: files/common/scripts/kind_provisioner.sh [hunk 2]
```diff
@@ -191,6 +191,44 @@ EOF
...
+  fi
+
```
Comment (jacob-delgado) [inline files/common/scripts/kind_provisioner.sh:231]:
```suggestion
```

File: files/common/scripts/kind_provisioner.sh (hunk 1 above)
Comment (Kmoneal) [reply files/common/scripts/kind_provisioner.sh]: Removed coredns because it's likely not needed

**Comment by Kmoneal** [comment]
@nmittler we would appreciate any feedback, if possible

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**